src/main/java/seedu/address/logic/commands/tagcommands/TagAddCommand.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.logic.commands.tagcommands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Adds a tag to the address book.
+ */
+public class TagAddCommand extends Command {
+
+    public static final String COMMAND_WORD = "tagadd";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a tag to the address book. "
+            + "Parameters: "
+            + PREFIX_TAG + "TAG\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_TAG + "cs2101";
+
+    public static final String MESSAGE_SUCCESS = "New tag added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This tag already exists in the address book";
```

Comment:
i think should be MESSAGE_DUPLICATE_TAG?

Reply:
yes thanks